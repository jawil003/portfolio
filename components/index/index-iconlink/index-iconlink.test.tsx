import { render } from "@testing-library/react";
import React from "react";
import {
  IndexIconLink,
  IndexIconLinkProps,
} from ".";

const variants: IndexIconLinkProps[] = [
  {
    size: "20px",
    href: "https://google.de",
    background: "white",
    external: true,
  },
  {
    size: "20px",
    href: "/test",
    background: "white",
  },
];

describe("IndexIconLink", () => {
  test.each(variants)(
    "Create IndexIconLink with %p",
    (variant) => {
      const testIconLink = render(
        // eslint-disable-next-line react/jsx-props-no-spreading
        <IndexIconLink {...variant} />,
      );

      expect(
        testIconLink.container
          .firstChild,
      ).toMatchSnapshot();
    },
  );

  test("Check styling ability of IconLink", () => {
    const testIconLink = render(
      <IndexIconLink
        // eslint-disable-next-line react/jsx-props-no-spreading
        {...variants[0]}
        className="testname"
      />,
    );

    expect(
      testIconLink.container.firstChild,
    ).toMatchSnapshot();
  });
});
