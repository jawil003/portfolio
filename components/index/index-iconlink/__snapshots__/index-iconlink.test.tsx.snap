// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexIconLink Check styling ability of IconLink 1`] = `
<body>
  <div>
    <a
      class="testname"
      href="https://google.de"
      rel="noreferrer"
      style="background-color: white; width: 20px; height: 20px; display: flex; justify-content: center; align-items: center; transform: none;"
      target="_blank"
    />
  </div>
</body>
`;

exports[`IndexIconLink Create IndexIconLink with {"background": "white", "external": true, "href": "https://google.de", "size": "20px"} 1`] = `
<body>
  <div>
    <a
      href="https://google.de"
      rel="noreferrer"
      style="background-color: white; width: 20px; height: 20px; display: flex; justify-content: center; align-items: center; transform: none;"
      target="_blank"
    />
  </div>
</body>
`;

exports[`IndexIconLink Create IndexIconLink with {"background": "white", "href": "/test", "size": "20px"} 1`] = `
<body>
  <div>
    <a
      style="background-color: white; width: 20px; height: 20px; display: flex; justify-content: center; align-items: center;"
    />
  </div>
</body>
`;
